src/main/java/com/iwaconsolti/league/demo/service/BasketLeague.java

```
@@ -0,0 +1,152 @@
+package com.iwaconsolti.league.demo.service;
+
+import com.iwaconsolti.league.demo.dto.MatchDTO;
+import com.iwaconsolti.league.demo.dto.PlayerDTO;
+import com.iwaconsolti.league.demo.dto.TeamDTO;
+import jakarta.annotation.PostConstruct;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import lombok.ToString;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Profile("populated")
+@Service("basketleague")
+@Slf4j
+@Getter
+@Setter
+@ToString
+@NoArgsConstructor
+public class BasketLeague implements LeagueInterface {
+    @Value("${league.teamLimit:10}")
+    private int TEAMLIMIT;
+
+    private final List<MatchDTO> matchDTOS = new ArrayList<>();
+    private final List<TeamDTO> teamDTOS = new ArrayList<>();
+    private final List<PlayerDTO> playerDTOS = new ArrayList<>();
+    private int ID;
+
+    public BasketLeague(int ID, int limit) {
+        this.ID = ID;
+    }
+
+    @Override
+    public void createTeam(TeamDTO teamDTO) {
+        if (this.teamDTOS.size() == TEAMLIMIT) {
+            log.error("You reached the max of team per league");
```

You should throw an exception or something so the user can get the feedback about the limit